src/sqlfluff/rules/L008.py

```diff
@@ -38,16 +38,24 @@ def _eval(self, segment, raw_stack, **kwargs):
         This is a slightly odd one, because we'll almost always evaluate from a point a few places
         after the problem site. NB: We need at least two segments behind us for this to work.
         """
+        if len(raw_stack) < 1:
+            return None
+
+        cm2 = raw_stack[-1]
+        if cm2.name == "comma":
+            # comma followed by something that isn't whitespace?
+            if segment.name not in ["whitespace", "newline"]:
+                ins = WhitespaceSegment(raw=" ")
+                return LintResult(
+                    anchor=cm2, fixes=[LintFix("edit", segment, [ins, segment])]
+                )
```

Comment:
Structure looks good. Nitpick: `cm2` refers to _something-minus-2_ and in this case we're `raw_stack[-1]`. We should either call it `cm1` or maybe even just refer to the element in the raw stack directly. The `cm2` here I think is confusing.

Arguably the `cm1` and `cm2`  in the part below aren't much clearer, but I don't think they _need_ to be refactored as part of this PR.

Reply:
Oh yes, I meant to do that. Will clean this up. It may have been possible to fix this without splitting the two cases, but it seemed good to avoid going any farther up `raw_stack` than necessary